lib/stateView/document/fetchDocumentsFactory.js

```diff
@@ -1,8 +1,14 @@
+const InvalidDocumentTypeError = require('@dashevo/dpp/lib/errors/InvalidDocumentTypeError');
+
+const InvalidQueryError = require('./errors/InvalidQueryError');
+const validateIndexedFields = require('./query/validateIndexedFields');
+const getIndexedFieldsFromDocumentSchema = require('./getIndexedFieldsFromDocumentSchema');
 /**
  * @param {createSVDocumentMongoDbRepository} createSVDocumentRepository
+ * @param {SVContractMongoDbRepository} svContractMongoDbRepository
  * @returns {fetchDocuments}
  */
-function fetchDocumentsFactory(createSVDocumentRepository) {
+function fetchDocumentsFactory(createSVDocumentRepository, svContractMongoDbRepository) {
```

Comment:
Should we pass `getIndexedFieldsFromDocumentSchema` and `validateIndexedFields` as dependency?

Reply:
Why? We use validators without defining them as dependencies in other places. As for `getIndexedFieldsFromDocumentSchema` - I think it's a part of this code.

Reply:
We follow DI despite the business logic and purposes of dependencies.